Src/CSharpier.Rider/src/main/java/com/intellij/csharpier/CSharpierProcessProvider.java

```diff
@@ -219,16 +219,26 @@ private ICSharpierProcess setupCSharpierProcess(String directory, String version
             // but there are still 0.12 and 0.14 downloads happening
             var installedVersion = version.split("\\.");
             var versionWeCareAbout = Integer.parseInt(installedVersion[1]);
+            var serverVersion = 29;
 
             ICSharpierProcess csharpierProcess;
-            if (versionWeCareAbout >= 28) {
+            if (versionWeCareAbout >= serverVersion
+                && !CSharpierSettings.getInstance(this.project).getDisableCSharpierServer()
+            ) {
                 csharpierProcess = new CSharpierProcessServer(customPath, version, this.project);
             } else if (versionWeCareAbout >= 12) {
                 var useUtf8 = versionWeCareAbout >= 14;
 
+                if (versionWeCareAbout >= serverVersion
+                    && CSharpierSettings.getInstance(this.project).getDisableCSharpierServer()
+                ) {
+                    this.logger.debug(
+                    "CSharpier server is disabled, falling back to piping via stdin"
```

Comment:
on an unrelated note, are you using an auto-formatter for Java? 
Otherwise, maybe this string should be indented?

Reply:
Every time I'm in there I wish that I was, I should finally figure out how to set one up